src/assert_equal.bash

```diff
@@ -31,12 +32,29 @@
 #   actual   : have
 #   --
 #   ```
+#
+# If the `--diff` option is set, a diff between the expected and actual output is shown.
 assert_equal() {
+  local -i show_diff=0
+
+  while (( $# > 0 )); do
+    case "$1" in
+    -d|--diff) show_diff=1; shift ;;
```

Comment:
This could be a breaking change for some users that want to check if something is `-d`/`--diff`

Unfortunately, we don't have a `--` interface established here. I think at minimum we should check that there are still (at least) two parameters left after the shift.